### PR TITLE
feat: optimize rss output format and add rss ignored support

### PR DESCRIPTION
**Pull request by pseudoyu**
I optimized the rss format accoring to hugo best practice and add "rss_ignore" prarm to manually
ignore some pages to to fetched by rss readers

### CODE REVIEW

File: layouts/rss.xml [hunk 1]
```diff
@@ -11,13 +11,16 @@
...
     <title>{{ if eq  .Title  .Site.Title }}{{ .Site.Title }}{{ else }}{{ with .Title }}{{.}} on {{ end }}{{ .Site.Title }}{{ end }}</title>
     <link>{{ .Permalink }}</link>
     <description>Recent content {{ if ne  .Title  .Site.Title }}{{ with .Title }}in {{.}} {{ end }}{{ end }}on {{ .Site.Title }}</description>
-    {{- with .Site.Params.customFavicon -}}
-    <webfeeds:icon>{{ . | absURL }}</webfeeds:icon>
+    {{- with .Site.Params.siteLogoImage -}}
+    <image>
+      <url>{{ . | absURL }}</url>
+      <link>{{ . | absURL }}</link>
+    </image>
```
Comment (shaform) [inline layouts/rss.xml]: This is probably not equivalent to  `<webfeeds:icon>` as it does not signal that it is an icon? The original syntax will allow RSS readers such as Feedly to use the specified image as the icon. See https://blog.haysc.tech/hexo-feed-setup/#%E8%87%AA%E5%88%B6-XML-%E6%A8%A1%E6%9D%BF%EF%BC%9A%E9%9D%A2%E5%90%91-Feedly-%E4%BC%98%E5%8C%96.

File: layouts/rss.xml [hunk 2]
```diff
@@ -11,13 +11,16 @@
 {{- $pages = $pages | first $limit -}}
 {{- end -}}
 {{- printf "<?xml version=\"1.0\" encoding=\"utf-8\" standalone=\"yes\"?>" | safeHTML }}
-<rss version="2.0" xmlns:atom="http://www.w3.org/2005/Atom" xmlns:webfeeds="http://webfeeds.org/rss/1.0">
+<rss version="2.0" xmlns:atom="http://www.w3.org/2005/Atom" xmlns:content="http://purl.org/rss/1.0/modules/content/">
```
Comment (shaform) [inline layouts/rss.xml]: To keep the ability to specify icon, we should not remove the webfeeds declaration.

File: layouts/rss.xml [hunk 3]
```diff
@@ -28,19 +31,18 @@
...
     {{ printf "<atom:link href=%q rel=\"self\" type=%q />" .Permalink .MediaType | safeHTML }}
     {{ end }}
-    {{ range $pages }}
+    {{ range $pages }}{{ if ne .Params.rss_ignore true }}
     <item>
       <title>{{ .Title }}</title>
       <link>{{ .Permalink }}</link>
       <pubDate>{{ .Date.Format "Mon, 02 Jan 2006 15:04:05 -0700" | safeHTML }}</pubDate>
-      {{ with .Site.Author.email }}<author>{{.}}{{ with $.Site.Author.name }} ({{.}}){{end}}</author>{{end}}
+      {{ with .Site.Author.email }}<author>{{.}}{{ with .Site.Author.name }} ({{.}}){{end}}</author>{{end}}
```
Comment (shaform) [inline layouts/rss.xml]: Mmmm... we originally have `$.Site.Author.name` because the default rss  template also have this. But it certainly seems to make sense to change this. Perhaps you could consider sending a PR to the main hugo repo and see what the authors say about this?

https://github.com/gohugoio/hugo/blob/65e52a7f5c79e6cfeabafa67d1bff4994c7fc4a4/tpl/tplimpl/embedded/templates/_default/rss.xml#L33

File: layouts/rss.xml [hunk 4]
```diff
@@ -28,19 +31,18 @@
...
       <guid>{{ .Permalink }}</guid>
-      {{ if .Site.Params.rssFullContent }}
-        <description>{{ .Content | html }}</description>
-      {{ else }}
-        <description>{{ .Summary | html }}</description>
-      {{ end }}
+      <description>{{ with .Description | html }}{{ . }}{{ else }}{{ .Summary | html }}{{ end -}}</description>
+      {{- if not .Site.Params.RSSNoContent}}
```
Comment (shaform) [inline layouts/rss.xml]: Could we make it backward-compatible and just use the original `.Site.Params.rssFullContent`?

**Review by shaform — CHANGES_REQUESTED** [review]
Thanks for your PR! However, I feel that your changes would remove some features. Please see the inline comments. Thanks!

File: layouts/rss.xml (hunk 4 above)
Comment (pseudoyu) [reply layouts/rss.xml]: Sure, I'll make the change. Thank you!

File: layouts/rss.xml (hunk 3 above)
Comment (pseudoyu) [reply layouts/rss.xml]: ok, I will do that. Before that, I temporarily revert the change.

File: layouts/rss.xml (hunk 4 above)
Comment (pseudoyu) [reply layouts/rss.xml]: > Could we make it backward-compatible and just use the original `.Site.Params.rssFullContent`?

I changed it and commit again.

File: layouts/rss.xml [hunk 5]
```diff
@@ -16,8 +16,7 @@
     <title>{{ if eq  .Title  .Site.Title }}{{ .Site.Title }}{{ else }}{{ with .Title }}{{.}} on {{ end }}{{ .Site.Title }}{{ end }}</title>
     <link>{{ .Permalink }}</link>
     <description>Recent content {{ if ne  .Title  .Site.Title }}{{ with .Title }}in {{.}} {{ end }}{{ end }}on {{ .Site.Title }}</description>
-    {{- with .Site.Params.customFavicon -}}
-    <webfeeds:icon>{{ . | absURL }}</webfeeds:icon>
+    {{- with .Site.Params.siteLogoImage -}}
```
Comment (shaform) [inline layouts/rss.xml]: Please also revert the changes here to keep the icon.

**Review by shaform — CHANGES_REQUESTED** [review]

File: layouts/rss.xml (hunk 5 above)
Comment (pseudoyu) [reply layouts/rss.xml]: ok, it's reverted. Please check, thank you!

File: layouts/rss.xml (hunk 3 above)
Comment (pseudoyu) [reply layouts/rss.xml]: https://github.com/gohugoio/hugo/pull/10065#issuecomment-1173594654

The Hugo maintainer said that using `$` makes no functional difference but will make it much clearer.

File: layouts/rss.xml [hunk 6]
```diff
@@ -11,16 +11,12 @@
 {{- $pages = $pages | first $limit -}}
 {{- end -}}
 {{- printf "<?xml version=\"1.0\" encoding=\"utf-8\" standalone=\"yes\"?>" | safeHTML }}
-<rss version="2.0" xmlns:atom="http://www.w3.org/2005/Atom" xmlns:content="http://purl.org/rss/1.0/modules/content/">
+<rss version="2.0" xmlns:atom="http://www.w3.org/2005/Atom" xmlns:webfeeds="http://webfeeds.org/rss/1.0">
```
Comment (shaform) [inline layouts/rss.xml]: You should keep both the `webfeeds` and `content` since you are also using `content` in your latest revision.

**Review by shaform — CHANGES_REQUESTED** [review]
Thanks for the changes! There is just one addition revision that has to be made. Please see my inline comment. Thanks!

File: layouts/rss.xml (hunk 6 above)
Comment (pseudoyu) [reply layouts/rss.xml]: Okay, I added the content part and push a commit. Thank you so much for your patience.

**Review by shaform — APPROVED** [review]
